src/gui/plugins/scene3d/Scene3D.cc

```diff
@@ -1607,10 +1607,53 @@ void IgnRenderer::HandleMouseTransformControl()
     else if (this->dataPtr->transformControl.Mode() ==
         rendering::TransformMode::TM_SCALE)
     {
+      // Check if the model that we're trying to scale looks like a simple shape
+      // It will need to have only two children nodes: one for the visual gizmo
+      // and another one for the link.
```

Comment:
How about just traversing down the tree (e.g. depth first or breadth first?) until you find a leaf node with the `geometry-type` UserData? Because we attach other objects to nodes, e.g. After doing `View` -> `Collisions`, new collision visuals are attached and the logic below will no longer work.

Reply:
Are you envisioning something like this? 00b7d93